### PR TITLE
Add the test for TakeDatasetOp

### PR DESCRIPTION
**Pull request by feihugis**
This PR adds the test for `TakeDatasetOp`.

cc: @jsimsa @rachellim

### CODE REVIEW

File: tensorflow/core/kernels/data/take_dataset_op_test.cc [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+          /*expected_outputs*/
+          {},
+          /*expected_output_dtypes*/ {DT_INT64},
+          /*expected_output_shapes*/
+          {PartialTensorShape({1})},
+          /*expected_cardinality*/ 0,
+          /*breakpoints*/ {0, 2, 5, 11}};
+}
+
+class ParameterizedDatasetTest
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/take_dataset_op_test.cc]: ParametrizedTakeDatasetOpTest

File: tensorflow/core/kernels/data/take_dataset_op_test.cc [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+  int64 count;
+  std::vector<Tensor> expected_outputs;
+  DataTypeVector expected_output_dtypes;
+  std::vector<PartialTensorShape> expected_output_shapes;
+  int64 expected_cardinality;
+  std::vector<int> breakpoints;
+};
+
+// Test case 1: take fewer than input size
+TestCase TakeFewerTestCase() {
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/take_dataset_op_test.cc]: TakeLessTestCase

**Review by jsimsa — CHANGES_REQUESTED** [review]
A couple of minor comments, but overall this looks good to me.

**Comment by jsimsa** [comment]
@feihugis Thank you for another unit test! I like that unit test style that we have converged on (defining test cases through functions that return a TestCase). When you get a chance, could you please apply it consistently across the existing unit tests? For instance, I recently made changes to tensor_slice_dataset_op_test.cc and sparse_tensor_slice_dataset_op_test.cc to fix an internal test failure and it would be great if you could apply your test style to those for consistency. Thanks again for the contributions!

**Comment by feihugis** [comment]
@jsimsa Thanks for your quick review! The class and function names have been updated in [this commit](https://github.com/tensorflow/tensorflow/pull/26632/commits/66b8ba465139aafe1c66e4e2f1aa2d1a1d044c85).

Sure, I will update the existing unit tests this week and keep the test style on the future PRs.

**Review by jsimsa — APPROVED** [review]

**Pull request merged**